C-extension compilation warnings

```
src/buildHistogram.c: In function ‘populate1DHist_’:
src/buildHistogram.c:18:19: warning: variable ‘last_bin’ set but not used [-Wunused-but-set-variable]
     unsigned int *last_bin;
                   ^~~~~~~~
src/buildHistogram.c: In function ‘populate1DHist’:
src/buildHistogram.c:38:44: warning: unused parameter ‘obj’ [-Wunused-parameter]
 static PyObject * populate1DHist(PyObject *obj, PyObject *args)
                                  ~~~~~~~~~~^~~
src/buildHistogram.c: At top level:
src/buildHistogram.c:70:5: warning: missing initializer for field ‘ml_flags’ of ‘PyMethodDef’ {aka ‘struct PyMethodDef’} [-Wmissing-field-initializers]
     {0,            0}                             /* sentinel */
     ^
In file included from miniconda3/envs/py311/include/python3.11/Python.h:65,
                 from src/buildHistogram.c:9:
miniconda3/envs/py311/include/python3.11/methodobject.h:57:17: note: ‘ml_flags’ declared here
     int         ml_flags;   /* Combination of METH_xxx flags, which mostly
                 ^~~~~~~~
```
```
src/computeMean.c: In function ‘computeMean’:
src/computeMean.c:78:9: warning: variable ‘status’ set but not used [-Wunused-but-set-variable]
     int status=0;
         ^~~~~~
src/computeMean.c:74:41: warning: unused parameter ‘obj’ [-Wunused-parameter]
 static PyObject * computeMean(PyObject *obj, PyObject *args)
                               ~~~~~~~~~~^~~
src/computeMean.c: At top level:
src/computeMean.c:107:5: warning: missing initializer for field ‘ml_flags’ of ‘PyMethodDef’ {aka ‘struct PyMethodDef’} [-Wmissing-field-initializers]
     {0,            0}                             /* sentinel */
     ^
In file included from miniconda3/envs/py311/include/python3.11/Python.h:65,
                 from src/computeMean.c:14:
miniconda3/envs/py311/include/python3.11/methodobject.h:57:17: note: ‘ml_flags’ declared here
     int         ml_flags;   /* Combination of METH_xxx flags, which mostly
                 ^~~~~~~~
```

--- a/.clang-format
+++ b/.clang-format
@@ -1,0 +1,18 @@
+---
+BasedOnStyle: Google
+Language: Cpp
+AlwaysBreakAfterReturnType: None
+AlwaysBreakAfterDefinitionReturnType: TopLevel
+BreakBeforeBraces: Stroustrup
+DerivePointerAlignment: false
+PointerAlignment: Right
+SortIncludes: false
+SortUsingDeclarations: false
+IndentWidth: 4
+TabWidth: 4
+UseTab: Never
+AlignConsecutiveMacros: true
+ColumnLimit: 79
+ReflowComments: true
+SpaceBeforeParens: ControlStatements
+...

--- a/src/buildHistogram.c
+++ b/src/buildHistogram.c
@@ -9,17 +9,16 @@
 #include <Python.h>
 #include "numpy/arrayobject.h"
 
-int populate1DHist_(float *image, int image_elements,
-		    unsigned int *histogram, int histogram_elements,
-                    float minValue, float maxValue, float binWidth)
+int
+populate1DHist_(float *image, int image_elements, unsigned int *histogram,
+                int histogram_elements, float minValue, float maxValue,
+                float binWidth)
 {
     int i, idx;
     float f, hist_edge, v;
-    unsigned int *last_bin;
 
     f = 1.0f / binWidth;
     hist_edge = minValue + binWidth * histogram_elements;
-    last_bin = histogram + (histogram_elements - 1);
 
     if (maxValue > hist_edge) {
         maxValue = hist_edge;
@@ -35,27 +34,34 @@ int populate1DHist_(float *image, int image_elements,
     return 1;
 }
 
-static PyObject * populate1DHist(PyObject *obj, PyObject *args)
+static PyObject *
+populate1DHist(PyObject *obj, PyObject *args)
 {
     PyObject *oimage, *ohistogram;
     PyArrayObject *image, *histogram;
     float minValue, maxValue, binWidth;
     int status;
 
-    if (!PyArg_ParseTuple(args,"OOfff:populate1DHist",&oimage,&ohistogram,&minValue,&maxValue,&binWidth))
-	    return NULL;
+    (void)obj;
 
-    image = (PyArrayObject *)PyArray_ContiguousFromObject(oimage, NPY_FLOAT32, 1, 2);
+    if (!PyArg_ParseTuple(args, "OOfff:populate1DHist", &oimage, &ohistogram,
+                          &minValue, &maxValue, &binWidth))
+        return NULL;
+
+    image = (PyArrayObject *)PyArray_ContiguousFromObject(oimage, NPY_FLOAT32,
+                                                          1, 2);
 
     if (!image) return NULL;
 
-	histogram = (PyArrayObject *)PyArray_ContiguousFromObject(ohistogram, NPY_UINT32, 1, 1);
+    histogram = (PyArrayObject *)PyArray_ContiguousFromObject(
+        ohistogram, NPY_UINT32, 1, 1);
 
     if (!histogram) return NULL;
 
-    status = populate1DHist_((float *)PyArray_DATA(image), PyArray_Size((PyObject*)image),
-			     (unsigned int *)PyArray_DATA(histogram), PyArray_Size((PyObject*)histogram),
-			     minValue, maxValue, binWidth);
+    status = populate1DHist_(
+        (float *)PyArray_DATA(image), PyArray_Size((PyObject *)image),
+        (unsigned int *)PyArray_DATA(histogram),
+        PyArray_Size((PyObject *)histogram), minValue, maxValue, binWidth);
 
     Py_XDECREF(image);
     Py_XDECREF(histogram);
@@ -63,29 +69,32 @@ static PyObject * populate1DHist(PyObject *obj, PyObject *args)
     return Py_BuildValue("i", status);
 }
 
-static PyMethodDef buildHistogram_methods[] =
-{
-    {"populate1DHist",  populate1DHist, METH_VARARGS,
-        "populate1Dhist(image, histogram, minValue, maxValue, binWidth)"},
-    {0,            0}                             /* sentinel */
+#pragma clang diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+static PyMethodDef buildHistogram_methods[] = {
+    {"populate1DHist", populate1DHist, METH_VARARGS,
+     "populate1Dhist(image, histogram, minValue, maxValue, binWidth)"},
+    {0, 0} /* sentinel */
 };
+#pragma GCC diagnostic pop
 
 static struct PyModuleDef moduledef = {
-  PyModuleDef_HEAD_INIT,
-  "buildHistogram",            /* m_name */
-  "C build histogram module",  /* m_doc */
-  -1,                          /* m_size */
-  buildHistogram_methods,      /* m_methods */
-  NULL,                        /* m_reload */
-  NULL,                        /* m_traverse */
-  NULL,                        /* m_clear */
-  NULL,                        /* m_free */
+    PyModuleDef_HEAD_INIT,
+    "buildHistogram",           /* m_name */
+    "C build histogram module", /* m_doc */
+    -1,                         /* m_size */
+    buildHistogram_methods,     /* m_methods */
+    NULL,                       /* m_reload */
+    NULL,                       /* m_traverse */
+    NULL,                       /* m_clear */
+    NULL,                       /* m_free */
 };
 
-PyMODINIT_FUNC PyInit_buildHistogram(void)
+PyMODINIT_FUNC
+PyInit_buildHistogram(void)
 {
-	PyObject* m;
+    PyObject *m;
     import_array();
-	m = PyModule_Create(&moduledef);
-	return m;
+    m = PyModule_Create(&moduledef);
+    return m;
 }

--- a/src/computeMean.c
+++ b/src/computeMean.c
@@ -14,9 +14,10 @@
 #include <Python.h>
 #include "numpy/arrayobject.h"
 
-int computeMean_(float *image, int nelements, float clipmin, float clipmax,
-                 int *numGoodPixels, float *mean, float *stddev,
-				 float *minValue, float *maxValue)
+int
+computeMean_(float *image, int nelements, float clipmin, float clipmax,
+             int *numGoodPixels, float *mean, float *stddev, float *minValue,
+             float *maxValue)
 {
     int i;
     float tmpMinValue, tmpMaxValue;
@@ -34,7 +35,7 @@ int computeMean_(float *image, int nelements, float clipmin, float clipmax,
     tmpMaxValue = (clipmax < image[0]) ? clipmax : image[0];
 
     for (i = 0; i < nelements; i++) {
-        if ( (image[i] >= clipmin) && (image[i] <= clipmax) ) {
+        if ((image[i] >= clipmin) && (image[i] <= clipmax)) {
             /* Find lowest value in the clipped image */
             if (image[i] <= tmpMinValue) {
                 tmpMinValue = image[i];
@@ -60,29 +61,33 @@ int computeMean_(float *image, int nelements, float clipmin, float clipmax,
     *maxValue = tmpMaxValue;
     *mean = (float)(sum / *numGoodPixels);
 
-    for (i=0; i < nelements; i++) {
-        if ( (image[i] >= *minValue) && (image[i] <= *maxValue) ) {
-            sumdiff = sumdiff + ( (image[i] - *mean) * (image[i] - *mean) );
-	    }
+    for (i = 0; i < nelements; i++) {
+        if ((image[i] >= *minValue) && (image[i] <= *maxValue)) {
+            sumdiff = sumdiff + ((image[i] - *mean) * (image[i] - *mean));
+        }
     }
 
-    *stddev = (float)sqrt( sumdiff  / (*numGoodPixels - 1));
+    *stddev = (float)sqrt(sumdiff / (*numGoodPixels - 1));
 
     return 1;
 }
 
-static PyObject * computeMean(PyObject *obj, PyObject *args)
+static PyObject *
+computeMean(PyObject *obj, PyObject *args)
 {
     PyObject *oimage;
     PyArrayObject *image;
-    int status=0;
     int numGoodPixels;
     float clipmin, clipmax, mean, stddev, minValue, maxValue;
 
-    if (!PyArg_ParseTuple(args,"Off:computeMean",&oimage, &clipmin, &clipmax))
-	    return NULL;
+    (void)obj;
 
-    image = (PyArrayObject *)PyArray_ContiguousFromObject(oimage, NPY_FLOAT32, 1, 2);
+    if (!PyArg_ParseTuple(args, "Off:computeMean", &oimage, &clipmin,
+                          &clipmax))
+        return NULL;
+
+    image = (PyArrayObject *)PyArray_ContiguousFromObject(oimage, NPY_FLOAT32,
+                                                          1, 2);
 
     if (!image) return NULL;
 
@@ -92,37 +97,42 @@ static PyObject * computeMean(PyObject *obj, PyObject *args)
     minValue = 0;
     maxValue = 0;
 
-    status = computeMean_((float *)PyArray_DATA(image), PyArray_Size((PyObject*)image),
-			  clipmin, clipmax,
-			  &numGoodPixels, &mean, &stddev, &minValue, &maxValue);
+    computeMean_((float *)PyArray_DATA(image), PyArray_Size((PyObject *)image),
+                 clipmin, clipmax, &numGoodPixels, &mean, &stddev, &minValue,
+                 &maxValue);
     Py_XDECREF(image);
 
-    return Py_BuildValue("iffff",numGoodPixels,mean,stddev,minValue,maxValue);
+    return Py_BuildValue("iffff", numGoodPixels, mean, stddev, minValue,
+                         maxValue);
 }
 
-static PyMethodDef computeMean_methods[] =
-{
-    {"computeMean",  computeMean, METH_VARARGS,
-        "computeMean(image, clipmin, clipmax, numGoodPixels, mean, stddev, minValue, maxValue)"},
-    {0,            0}                             /* sentinel */
+#pragma clang diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+static PyMethodDef computeMean_methods[] = {
+    {"computeMean", computeMean, METH_VARARGS,
+     "computeMean(image, clipmin, clipmax, numGoodPixels, mean, stddev, "
+     "minValue, maxValue)"},
+    {0, 0} /* sentinel */
 };
+#pragma GCC diagnostic pop
 
 static struct PyModuleDef moduledef = {
-  PyModuleDef_HEAD_INIT,
-  "computeMean",           /* m_name */
-  "C compute mean module", /* m_doc */
-  -1,                      /* m_size */
-  computeMean_methods,     /* m_methods */
-  NULL,                    /* m_reload */
-  NULL,                    /* m_traverse */
-  NULL,                    /* m_clear */
-  NULL,                    /* m_free */
+    PyModuleDef_HEAD_INIT,
+    "computeMean",           /* m_name */
+    "C compute mean module", /* m_doc */
+    -1,                      /* m_size */
+    computeMean_methods,     /* m_methods */
+    NULL,                    /* m_reload */
+    NULL,                    /* m_traverse */
+    NULL,                    /* m_clear */
+    NULL,                    /* m_free */
 };
 
-PyMODINIT_FUNC PyInit_computeMean(void)
+PyMODINIT_FUNC
+PyInit_computeMean(void)
 {
-	PyObject* m;
-	import_array();
-	m = PyModule_Create(&moduledef);
-	return m;
+    PyObject *m;
+    import_array();
+    m = PyModule_Create(&moduledef);
+    return m;
 }

